src/Symfony/Component/HttpFoundation/JsonResponse.php

```diff
@@ -43,6 +43,10 @@ public function __construct($data = null, int $status = 200, array $headers = []
     {
         parent::__construct('', $status, $headers);
 
+        if ($json && !\is_string($data) && !is_numeric($data) && !\is_callable([$data, '__toString'])) {
+            throw new \TypeError(sprintf('The data must be a string or object implementing __toString(), "%s" given.', \get_debug_type($data)));
```

Comment:
We should include the method and the reason for the stricter type check in the error message. After all, it's the message that we want to improve here. How about this one?

```suggestion
            throw new \TypeError(sprintf('%s: If $json is set to true, argument $data must be a string or object implementing __toString(), "%s" given.', __METHOD__, \get_debug_type($data)));
```